src/python/nimbusml/pipeline.py

```diff
@@ -2073,13 +2073,9 @@ def _predict(self, X, y=None,
                 isinstance(X, DataFrame) and isinstance(y, (str, tuple))):
             y = y_temp
 
-        is_transformer_chain = False
-        with ZipFile(self.model) as model_zip:
-            is_transformer_chain = any('TransformerChain' in item
-                                   for item in model_zip.namelist())
-
         all_nodes = []
-        if is_transformer_chain:
+        if (hasattr(self, '_is_transformer_chain') and
+            self._is_transformer_chain):
             inputs = dict([('data', ''), ('transform_model', self.model)])
```

Comment:
How much is perf gain? If its not much I would like to leave this as it is. Reason is that we actually should move to new ML.NET format so then this will be broken with your new fix

Reply:
6% gain, with `.predict` called on 100 row UCI Adult test data, in a 100 rep for loop, repeated 5 times.

I think we should take the change. (1) If we leave it as it is, it will still be broken when NimbusML moves to new ML.NET format, and how the graph is constructed will need to change anyway. (2) Moving to new ML.NET format will likely take a long time, as it requires non-trivial changes to how model loading is handled in the entrypoints infrastructure on ML.NET side i.e. new model implementation in addition to `PredictorModel` and `TransformModel`. (3) The changes directly below this address an issue where `PredictedLabel` column is only converted to int32 from bool if it exists (i.e. it is not regression or ranking) and if the dtype is bool (i.e. if it is Binary classification).